CMakeLists.txt

```diff
@@ -39,12 +39,13 @@ include(GNUInstallDirs)
 #-----------------------------------------------------------------------------
 set(CMAKE_C_FLAGS_DEBUG "${CMAKE_C_FLAGS_DEBUG} -g3 -O0 -Wall -Wextra -Wno-missing-field-initializers -Wno-missing-braces -Wno-unused-result -Wno-ignored-attributes")
 set(CMAKE_C_FLAGS_RELEASE "${CMAKE_C_FLAGS_RELEASE} -Werror -O2")
+string(REGEX REPLACE "O2 " "O3 " CMAKE_C_FLAGS_RELWITHDEBINFO "${CMAKE_C_FLAGS_RELWITHDEBINFO} -g -ggdb3" )
 
 # default hardening flags that have no performance hit
 set(HARDENING_LINKER_FLAGS "-Wl,-z,noexecstack -Wl,-z,relro -Wl,-z,now")
 # update default flags
-set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} ${HARDENING_LINKER_FLAGS}")
-set(CMAKE_EXE_LINKER_FLAGS "${CMAKE_EXE_LINKER_FLAGS} ${HARDENING_LINKER_FLAGS}")
+set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} ${HARDENING_LINKER_FLAGS} -flto")
```

Comment:
CMake supports the target property `INTERPROCEDURAL_OPTIMIZATION` for enabling LTO. https://cmake.org/cmake/help/latest/prop_tgt/INTERPROCEDURAL_OPTIMIZATION.html
There is also a module for checking whether LTO is supported. https://cmake.org/cmake/help/latest/module/CheckIPOSupported.html

Reply:
Great! Thanks!

Reply:
I failed to use this option. I've tried something like this:
```
set(CMAKE_INTERPROCEDURAL_OPTIMIZATION TRUE)
```

But in build log there was no `-flto` options.

Reply:
@skvl I have never tried the global approach myself, but I was able to make it work as a target property.  I think this is the cleaner approach anyways.